### PR TITLE
Ruby 2.7.2

### PR DESCRIPTION
**Pull request by jmartin-r7**
* Ruby update to 2.7.2
* rubygems update to 3.1.4
* winpcap API version refactor for easier update

### CODE REVIEW

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 tests this please.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

File: config/software/ruby.rb [hunk 1]
```diff
@@ -170,51 +171,29 @@
...
-  # Backporting a 2.6.0 fix to 2.5.1 (and 2.4.4 for ChefDK 2). This allows us to build Nokogiri 1.8.3.
-  # Basically we only include `-Werror` linker warnings when building native gems if we are on Windows.
-  # This prevents some "expected" warnings from failing the build.
-  if version == "2.5.1" || version == "2.4.4"
-    patch source: "ruby-only-compiler-warnings-on-windows.patch", plevel: 1, env: patch_env
-  end
-
   configure_command = ["--with-out-ext=dbm,readline",
                        "--enable-shared",
-                       "--enable-load-relative",
```
Comment (adfoster-r7) [inline config/software/ruby.rb:191]: @jmartin-r7 This was enabled intentionally with enabling portable builds: https://github.com/rapid7/metasploit-omnibus/pull/127

Commit:
https://github.com/rapid7/metasploit-omnibus/commit/4e4904663045a18537e1ddd149d18a780db64a3d#diff-8ff31cf08a88774679ed15595ec42f8fea9608671ca50be341953489045e8e25R186

Unless enabled-load-relative is enabled by default in Ruby 2.7.2, I think we'll have to keep this line in 👀 

I can't think of a cleaner way to ensure this doesn't get accidentally deleted again in the future - other than adding a quick test for the configuration value being set maybe?

Comment (jmartin-r7) [reply config/software/ruby.rb:191]: I will restore, I must have missed this when porting in the latest upstream version of the file.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this plesae.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

**Comment by adfoster-r7** [comment]
Tested on windows 10 with a few modules, everything looks good to me :+1:

image

**Review by adfoster-r7 — APPROVED** [review]

**Pull request merged**